### PR TITLE
Android refactoring

### PR DESCRIPTION
**Pull request by albexk**

### CODE REVIEW

File: client/android/cloak/src/main/kotlin/Cloak.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        val configStr = "$openVpnConfigStr\n<cloak>\n$cloakConfigStr\n</cloak>\n"
+
+        openVpnConfig.usePluggableTransports = true
+        openVpnConfig.content = configStr
+        return openVpnConfig
+    }
+
+    private fun checkCloakJson(cloakConfigJson: JSONObject): JSONObject {
+        // todo: strange method
+        if (!cloakConfigJson.has("NumConn")) cloakConfigJson.put("NumConn", 1)
```
[the remark this answers is not in the document]
Comment (albexk) [reply client/android/cloak/src/main/kotlin/Cloak.kt:56]: Are the NumConn and ProxyMethod parameters mandatory or optional? Can their values be different? Do I need to check their presence to overwrite them, or can I just always add them to json with the required values?

**Review by Noxius87 — APPROVED** [review]

File: client/android/cloak/src/main/kotlin/Cloak.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (outspace) [reply client/android/cloak/src/main/kotlin/Cloak.kt:56]: > Are the NumConn and ProxyMethod parameters mandatory or optional?

These are mandatory parameters. 

> Can their values be different? 

For Android OpenVPN over Claok protocol -- no. "NumConn" should always be 1 (without brackets). ProxyMethod -- "openvpn".

> Do I need to check their presence to overwrite them, or can I just always add them to json with the required values?

You can just overwrite them in cloak config without chacking original values.

File: client/android/cloak/src/main/kotlin/Cloak.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        val configStr = "$openVpnConfigStr\n<cloak>\n$cloakConfigStr\n</cloak>\n"
+
+        openVpnConfig.usePluggableTransports = true
+        openVpnConfig.content = configStr
+        return openVpnConfig
+    }
+
+    private fun checkCloakJson(cloakConfigJson: JSONObject): JSONObject {
+        cloakConfigJson.put("NumConn", 1)
```
Comment (outspace) [inline client/android/cloak/src/main/kotlin/Cloak.kt:55]: Looks like you added these values into Cloak config and this field can be double included into Cloak config. 
I requested you overwrite these values in Cloak config, in previous review.

Comment (albexk) [reply client/android/cloak/src/main/kotlin/Cloak.kt:55]: The 'put' function adds a record to the json or overwrites it by key. The keys are not doubled.

**Pull request merged**